<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainForm</class>
 <widget class="QWidget" name="mainForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>425</width>
    <height>516</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Qt Echo Server</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout" stretch="0,0,1,0">
   <item>
    <widget class="QLabel" name="configurationLabel">
     <property name="text">
      <string>Configuration</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTabWidget" name="configTabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="executionTab">
      <attribute name="title">
       <string>Execution</string>
      </attribute>
      <layout class="QFormLayout" name="executionTabLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="sessionManagerThreadsLabel">
         <property name="text">
          <string>Session management threads</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="sessionManagerThreadsSpinBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
         <property name="value">
          <number>2</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="sessionThreadsLabel">
         <property name="text">
          <string>Session (IO) threads</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="sessionThreadsSpinBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
         <property name="value">
          <number>2</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="sessionThreadModelLabel">
         <property name="text">
          <string>Session threading model</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="sessionThreadModelComboBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <item>
          <property name="text">
           <string>Single demultiplexer</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Demultiplexer per work thread</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="sessionManagementTab">
      <attribute name="title">
       <string>Session management</string>
      </attribute>
      <layout class="QFormLayout" name="sessionManagementTabLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="maxSessionCountLabel">
         <property name="text">
          <string>Maximum active sessions (TCP connections)</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="maxSessionCountSpinBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1000000</number>
         </property>
         <property name="value">
          <number>10000</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="recycledSessionCountLabel">
         <property name="text">
          <string>Maximum recycled sessions</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="recycledSessionCountSpinBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>1000000</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="listenBacklogLabel">
         <property name="text">
          <string>Listen backlog</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QSpinBox" name="listenBacklogSpinBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
         <property name="value">
          <number>6</number>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="portNumberLabel">
         <property name="text">
          <string>TCP port number to listen</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QSpinBox" name="portNumberSpinBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>65535</number>
         </property>
         <property name="value">
          <number>7</number>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="addressLabel">
         <property name="text">
          <string>IP address of interface to listen (IPv4 or IPv6)</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLineEdit" name="addressEdit">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>0.0.0.0</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="sessionTab">
      <attribute name="title">
       <string>Session</string>
      </attribute>
      <layout class="QFormLayout" name="sessionTabLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="sessionBufferSizeLabel">
         <property name="text">
          <string>Session's buffer size (bytes)</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="sessionBufferSizeSpinBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>10485760</number>
         </property>
         <property name="singleStep">
          <number>256</number>
         </property>
         <property name="value">
          <number>4096</number>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QCheckBox" name="sockRecvBufferSizeCheckBox">
         <property name="text">
          <string>Socket's recieve buffer size (bytes)</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QSpinBox" name="sockRecvBufferSizeSpinBox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>10485760</number>
         </property>
         <property name="singleStep">
          <number>256</number>
         </property>
         <property name="value">
          <number>4096</number>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QCheckBox" name="sockSendBufferSizeCheckBox">
         <property name="text">
          <string>Socket's send buffer size (bytes)</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QSpinBox" name="sockSendBufferSizeSpinBox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>10485760</number>
         </property>
         <property name="singleStep">
          <number>256</number>
         </property>
         <property name="value">
          <number>4096</number>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="tcpNoDelayLabel">
         <property name="text">
          <string>TCP NO_DELAY option</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QComboBox" name="tcpNoDelayComboBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>default (operating system)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>on</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>off</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QCheckBox" name="inactivityTimeoutCheckBox">
         <property name="text">
          <string>Inactivity timeout (seconds)</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="inactivityTimeoutSpinBox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>10485760</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
         <property name="value">
          <number>600</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="maxTransferSizeLabel">
         <property name="text">
          <string>Max size of single async transfer (bytes)</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QSpinBox" name="maxTransferSizeSpinBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>10485760</number>
         </property>
         <property name="singleStep">
          <number>32</number>
         </property>
         <property name="value">
          <number>4096</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QTabWidget" name="stateTabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="statisticsTab">
      <attribute name="title">
       <string>Statistics</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="activeSessionsLabel">
         <property name="text">
          <string>Active sessions</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="activeSessionsEdit">
         <property name="text">
          <string>0</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="maxActiveSessionsLabel">
         <property name="text">
          <string>Maximum of active sessions</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="maxActiveSessionsEdit">
         <property name="text">
          <string>0</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="recycledSessionsLabel">
         <property name="text">
          <string>Recycled sessions</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="recycledSessionsEdit">
         <property name="text">
          <string>0</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="totalAcceptedSessionsLabel">
         <property name="text">
          <string>Total accepted sessions</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="totalAcceptedSessionsEdit">
         <property name="text">
          <string>0</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="activeShutdownedSessionsLabel">
         <property name="text">
          <string>Active shutdowned sessions</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLineEdit" name="activeShutdownedSessionsEdit">
         <property name="text">
          <string>0</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="outOfWorkSessionsLabel">
         <property name="text">
          <string>Passive shutdowned sessions</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QLineEdit" name="outOfWorkSessionsEdit">
         <property name="text">
          <string>0</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="timedOutSessionsLabel">
         <property name="text">
          <string>Timed out sessions</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QLineEdit" name="timedOutSessionsEdit">
         <property name="text">
          <string>0</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="errorStoppedSessionsLabel">
         <property name="text">
          <string>Error stopped sessions</string>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QLineEdit" name="errorStoppedSessionsEdit">
         <property name="text">
          <string>0</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="logTab">
      <attribute name="title">
       <string>Log</string>
      </attribute>
      <layout class="QVBoxLayout" name="logTabLayout">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="QPlainTextEdit" name="logTextEdit">
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="tabChangesFocus">
          <bool>true</bool>
         </property>
         <property name="lineWrapMode">
          <enum>QPlainTextEdit::NoWrap</enum>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="controlButtonLayout" stretch="0,0,0,0,0,0">
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="text">
        <string>&amp;Clear log</string>
       </property>
       <property name="shortcut">
        <string>Ctrl+L</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="controlButtonSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="startButton">
       <property name="text">
        <string>&amp;Start</string>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="stopButton">
       <property name="text">
        <string>Sto&amp;p</string>
       </property>
       <property name="shortcut">
        <string>Ctrl+S</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="terminateButton">
       <property name="text">
        <string>&amp;Terminate</string>
       </property>
       <property name="shortcut">
        <string>Ctrl+T</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="exitButton">
       <property name="text">
        <string>E&amp;xit</string>
       </property>
       <property name="shortcut">
        <string>Ctrl+Q</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>configTabWidget</tabstop>
  <tabstop>sessionManagerThreadsSpinBox</tabstop>
  <tabstop>sessionThreadsSpinBox</tabstop>
  <tabstop>sessionThreadModelComboBox</tabstop>
  <tabstop>maxSessionCountSpinBox</tabstop>
  <tabstop>recycledSessionCountSpinBox</tabstop>
  <tabstop>listenBacklogSpinBox</tabstop>
  <tabstop>portNumberSpinBox</tabstop>
  <tabstop>addressEdit</tabstop>
  <tabstop>sessionBufferSizeSpinBox</tabstop>
  <tabstop>inactivityTimeoutCheckBox</tabstop>
  <tabstop>inactivityTimeoutSpinBox</tabstop>
  <tabstop>maxTransferSizeSpinBox</tabstop>
  <tabstop>sockRecvBufferSizeCheckBox</tabstop>
  <tabstop>sockRecvBufferSizeSpinBox</tabstop>
  <tabstop>sockSendBufferSizeCheckBox</tabstop>
  <tabstop>sockSendBufferSizeSpinBox</tabstop>
  <tabstop>tcpNoDelayComboBox</tabstop>
  <tabstop>stateTabWidget</tabstop>
  <tabstop>activeSessionsEdit</tabstop>
  <tabstop>maxActiveSessionsEdit</tabstop>
  <tabstop>recycledSessionsEdit</tabstop>
  <tabstop>totalAcceptedSessionsEdit</tabstop>
  <tabstop>activeShutdownedSessionsEdit</tabstop>
  <tabstop>outOfWorkSessionsEdit</tabstop>
  <tabstop>timedOutSessionsEdit</tabstop>
  <tabstop>errorStoppedSessionsEdit</tabstop>
  <tabstop>logTextEdit</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>startButton</tabstop>
  <tabstop>stopButton</tabstop>
  <tabstop>terminateButton</tabstop>
  <tabstop>exitButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>logTextEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>385</y>
    </hint>
    <hint type="destinationlabel">
     <x>57</x>
     <y>321</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sockRecvBufferSizeCheckBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>sockRecvBufferSizeSpinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>170</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel">
     <x>402</x>
     <y>130</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sockSendBufferSizeCheckBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>sockSendBufferSizeSpinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>402</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exitButton</sender>
   <signal>clicked()</signal>
   <receiver>mainForm</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>362</x>
     <y>376</y>
    </hint>
    <hint type="destinationlabel">
     <x>368</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>inactivityTimeoutCheckBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>inactivityTimeoutSpinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>98</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>237</x>
     <y>98</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
